RevenueCatUI/UIKit/PaywallFooterViewController.swift

```diff
@@ -34,15 +34,17 @@ public final class PaywallFooterViewController: PaywallViewController {
     public init(offering: Offering? = nil) {
         super.init(content: .optionalOffering(offering),
                    fonts: DefaultPaywallFontProvider(),
-                   displayCloseButton: false)
+                   displayCloseButton: false,
+                   dismissalRequest: nil)
```

Comment:
Hmm I think we should still accept a dismissalRequest parameter for the footer, since there might be a successful purchase? Alternatively, we can just devs use the purchaseCompleted callback, which would be called in the same location... But not sure if we want to be consistent with that behavior between the full screen and footer.

Reply:
hmmm, yes. I think it makes sense, I will add it to `PaywallFooterViewController`

Reply:
Ok... so I added it, but the callback is not being called because there's this code in `PaywallView`

```
              if self.mode.isFullScreen, purchased {
                    Logger.debug(Strings.dismissing_paywall)
                    guard let onRequestedDismissal = self.onRequestedDismissal else {
                        self.dismiss()
                        return
                    }
                    onRequestedDismissal()
                }
```

I am going to see what happens in Android... but if we don't call the `onDismiss` in Android, I think we shouldn't do it here either